vmm/src/lib.rs

```diff
@@ -1372,9 +1404,11 @@ impl SendAdditionalConnections {
 impl Drop for SendAdditionalConnections {
     fn drop(&mut self) {
         info!("Sending disconnect message to channels");
-        self.channels
-            .drain(..)
-            .for_each(|channel| channel.send(SendMemoryThreadMessage::Disconnect).unwrap());
+        self.channels.drain(..).for_each(|channel| {
```

Comment:
I don't have a strong opinon here but how about some form of

```suggestion
    if self.cancel.load(Ordering::Acquire) {
        self.channels.drain(..).for_each(|channel| {
            let _ = channel.send(SendMemoryThreadMessage::Disconnect);
        };
    else {
        self.channels
            .drain(..)
            .for_each(|channel| channel.send(SendMemoryThreadMessage::Disconnect).unwrap());
    }
```

Reply:
I can do that, but I do not really see what we would win with that. Can you elaborate?

Reply:
We are loosing errors returned by .send(). 

But probably you also could just do

```
let e = channel.send(SendMemoryThreadMessage::Disconnect);
if let Err(e) = e {
    log::error!("..... {e}")
}
```

Reply:
Fixed.